Remove Wagtail dependency, add typing, add pre-commit.

olivierphi opened the pull request:
In a nutshell:  
 - Remove Wagtail dependency, add type hint and declare the package as being "officially" typed via the `py.typed` file, add `pre-commit`.
 - Add BucketAVMixin.  
    This class comes straight from Jake's Merge Request: https://github.com/torchbox/wagtail-bucketav/pull/5/files
 - Update Django<->Python test matrix.
     I've left Django 3.2, but as it go EOL in less than 3 months we could also just decide to not support it in this package? 
 - CI is green: https://github.com/olivierphi/wagtail-bucketav/actions/runs/7488149279 :heavy_check_mark: 

## TODO?

 - Rename the package from `wagtail-bucketav` to `django-bucketav`?
 - Add some Wagtail-specific :bird: code in a `contrib.wagtail` sub-package? What do you folks think such a Wagtail specialisation should contain?
 - As we're not using MkDocs for the moment, get rid of the `documentation` dependencies? (they are the heaviest ones)

Commit:
Remove Wagtail dependency, add typing, add pre-commit.

Commit:
Add BucketAVMixin

This class comes from Jake's Merge Request: https://github.com/torchbox/wagtail-bucketav/pull/5/files

Commit:
[CI] Update Django<->Python test matrix

olivierphi commented on .flake8:
```diff
@@ -0,0 +1,2 @@
+[flake8]
+max-line-length = 88
```
Since we're using Black with the default 88 chars max line length, we should set up Flake8 accordingly 🙂

(note that we could also replace Flake8 and isort with Ruff while we're there? Happy to do it if you folks think it makes sense)

olivierphi commented on .github/workflows/test.yml:
```diff
@@ -23,9 +23,9 @@ jobs:
-          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
-          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+          flake8 wagtail_bucketav --count --select=E9,F63,F7,F82 --show-source --statistics
```
I reckon a next step would be to rename that folder (and the repo) so something like `django_bucketav` ? 🙂

RealOrangeOne replied:
Yep, that works. I think that should be as part of this PR.

olivierphi commented on pyproject.toml:
```diff
@@ -20,38 +20,37 @@ classifiers = [
 lint = [
     "black==22.6.0",
-    "isort==5.10.1",
+    "isort==5.11.5",
```
had to upgrade to that version to make it "pre-commit"-proof

olivierphi commented on pyproject.toml:
```diff
@@ -20,38 +20,37 @@ classifiers = [
 ]
 test = [
     "pytest-cov==3.0.0",
-    "pytest-mock==3.8.2",
```
we were not actually using this package :slightly_smiling_face:

olivierphi commented on wagtail_bucketav/settings.py:
```diff
@@ -1,14 +1,15 @@
-    get_document_model_string(): ["file"],
-}
+if TYPE_CHECKING:
+    from typing_extensions import Dict, List, TypeAlias
```
have to use these "old school type hints" since we're supporting Python 3.8

olivierphi commented on wagtail_bucketav/tests/conftest.py:
```diff
@@ -31,3 +33,25 @@ def scan_result_received_receiver():
+
+
+@pytest.fixture
+def django_file() -> ContentFile:
```
Simplified version of what FactoryBoy's Django FileField factory does under the hood 🙂

olivierphi commented on wagtail_bucketav/tests/test_utils.py:
```diff
@@ -1,24 +1,13 @@
 
 @pytest.mark.django_db
-def test_get_image_with_key():
-    image = wagtail_factories.ImageFactory()
```
We can just inject the test models via Pytest's dependency injection machinery 🙂

olivierphi commented on wagtail_bucketav/urls.py:
```diff
@@ -4,4 +4,6 @@
 app_name = "wagtail_bucketav"
 
-urlpatterns = [path("sns-hook/", BucketAVWebhookView.as_view(), name="sns-hook")]
+urlpatterns = [
```
more readable when each URL pattern lives on its own line?  
(but that's very subjective, happy to revert it of course 🙂 )

olivierphi commented on wagtail_bucketav/views.py:
```diff
@@ -11,12 +11,12 @@
-    def handle_message(self, message, payload):
-        message = json.loads(message)  # Actually parse the JSON
+    def handle_message(self, message: str, payload) -> None:
+        message_dict = json.loads(message)  # Actually parse the JSON
```
[while I was there] let's avoid re-using the same message variable as a string and then as a dict? :slightly_smiling_face:

RealOrangeOne commented on .github/workflows/test.yml:
```diff
@@ -38,19 +38,35 @@ jobs:
-            python-version: 3.9
+          # Django 5.x supports Python 3.10, 3.11, and 3.12
+          # https://docs.djangoproject.com/en/5.0/releases/5.0/#python-compatibility
+          - django-version: "Django>=5.0,<6.0"
```
Suggestion: Should we explicitly limit to 5.0.x only? 5.1 isn't going to be for a while, but still.

olivierphi replied:
You're right! Addressed in [3e9ab2a](https://github.com/torchbox/wagtail-bucketav/pull/6/commits/3e9ab2a1f72af7931b787744741041623d2150bb)

RealOrangeOne commented on wagtail_bucketav/models.py:
```diff
@@ -7,3 +8,21 @@ class FileScanStatus(models.TextChoices):
+    def update_bucketav_scan_status(self, scan_status: FileScanStatus) -> None:
+        self.bucketav_scan_status = scan_status
+        self.bucketav_last_scanned_at = timezone.now()
+        self.save()
```
Suggestion: Add `update_fields` here so it only updates the values we want
```suggestion
        self.save(update_fields=['bucketav_scan_status', 'bucketav_last_scanned_at'])
```

olivierphi replied:
Better indeed! Addressed in [3e9ab2a](https://github.com/torchbox/wagtail-bucketav/pull/6/commits/3e9ab2a1f72af7931b787744741041623d2150bb)

Commit:
Address code review feedback

The most important change is the renaming of the package, from `wagtail-bucketav` to `django-bucketav`.

Commit:
Add a quick documentation

This will have to improved for sure, but that's a first step already :-)

RealOrangeOne commented on django_bucketav/settings.py:
```diff
@@ -12,4 +12,4 @@
 
 def bucketav_models() -> "BucketAVModelDefinition":
-    return getattr(settings, "WAGTAIL_BUCKETAV_MODELS", {})
+    return getattr(settings, "DJANGO_BUCKETAV_MODELS", {})
```
Suggestion: Given it's implied that settings are for Django, perhaps this should be `BUCKETAV_MODELS`?

olivierphi replied:
Good call! :slightly_smiling_face:  Handled in [b6d7f1c](https://github.com/torchbox/wagtail-bucketav/pull/6/commits/b6d7f1cca851c831b4a14ed8d86a41103598b23b)

Commit:
The Django settings object is now `BUCKETAV_MODELS`

We know we're in Django right? So no need to prefix the settings name with `DJANGO_` 😅
Thanks Jake for the suggestion :-)
